src/gluonts/dataset/common.py

```diff
@@ -315,10 +315,12 @@ def __call__(self, data: DataEntry) -> DataEntry:
 
     @staticmethod
     @lru_cache(maxsize=10000)
-    def process(string: str, freq: str) -> pd.Timestamp:
+    def process(
+        string_or_ts: Union[str, pd.Timestamp], freq: str
```

Comment:
In theory, a lot of different values should work here. Maybe we just use `Any` here with a comment indicating such?

Reply:
sure sounds good to me! I checked the pandas docs and it supports: `datetime-like, str, int, float` types...